demo/web/src/main/notebooks/Python_AI/02 Deephaven with TensorFlow.md

```diff
@@ -0,0 +1,200 @@
+    outputs = [learn.Output("PredictedClass", numpy_to_table_integer, "int")],
+    batch_size = iris_test.intSize()
+```
+\
+\
+The model works like a charm!  We classified the Iris flower dataset using Deephaven tables and TensorFlow.  That's kind of cool, but it's cooler to do classifications on a live feed of incoming observations.  We'll now show how to do that.  As you'll see, it turns out it's incredibly easy to do so with Deephaven!
+
+## Define bounds for making realistic faux Iris measurements
+
+We first need to set up a table that we can write faux Iris measurements in real-time to.  We can use the minimum and maximum of each measurement to make sure the faux measurements are realistic.
+
+```python
+min_petal_length, max_petal_length = raw_iris["PetalLengthCM"].min(), raw_iris["PetalLengthCM"].max()
+min_petal_width, max_petal_width = raw_iris["PetalWidthCM"].min(), raw_iris["PetalWidthCM"].max()
+min_sepal_length, max_sepal_length = raw_iris["SepalLengthCM"].min(), raw_iris["SepalLengthCM"].max()
+min_sepal_width, max_sepal_width = raw_iris["SepalWidthCM"].min(), raw_iris["SepalWidthCM"].max()
+```
+
+## Write the faux measurements to a table in real-time
+
+With these quantities now calculated and stored in memory, we need to set up a live table that we can write faux measurements to.  To keep it simple, we'll write measurements to the table once per second for a minute.
+
+```python
+table_writer = DynamicTableWriter(
+    ["SepalLengthCM", "SepalWidthCM", "PetalLengthCM", "PetalWidthCM"],
+    [dht.double] * 4
+)
+
+live_iris = table_writer.getTable()
+
+def write_faux_iris_measurements():
+    for i in range(60):
+        sepal_length = np.round(random.uniform(min_sepal_length, max_sepal_length), 1)
+        sepal_width = np.round(random.uniform(min_sepal_width, max_sepal_width), 1)
+        petal_length = np.round(random.uniform(min_petal_length, max_petal_length), 1)
+        petal_width = np.round(random.uniform(min_petal_width, max_petal_width), 1)
+
+        table_writer.logRow(sepal_length, sepal_width, petal_length, petal_width)
+        time.sleep(1)
+
+thread = threading.Thread(target = write_faux_iris_measurements)
+thread.start()
+```
+
+## Use the trained model on live data
+
+Now we've got some faux live incoming measurements.  We can just use the model we've already trained with the functions we've already created to do live classification!
+
+```python
+iris_classifications_live = learn.learn(
+    table = live_iris,
+    model = use_trained_model,
+    inputs = [learn.Input(["SepalLengthCM", "SepalWidthCM", "PetalLengthCM", "PetalWidthCM"], table_to_numpy_double)],
+    outputs = [learn.Output("PredictedClass", numpy_to_table_integer, "int")],
+    batch_size = 5
+)
+```
+
+And that's all there is to it.  The only extra work we needed to do in order to use our model on real-time data is set up the real-time data itself.  This might be a simple problem, but this holds true for more complex problems.
```

Comment:
```suggestion
That's all there is to it.  The only extra work required is to set up the real-time data itself.  This might be a simple problem, but this holds true for more complex problems.
```